backend/routes/post.py

```
@@ -294,32 +302,36 @@ async def create_event_post(
             os.makedirs(upload_folder, exist_ok=True)
 
-            file_extension = os.path.splitext(event_image.filename)[1]
-            file_name = f"{uuid4()}{file_extension}"
+            # Generate a unique filename using uuid to avoid path injection attacks
+            file_name = sanitize_filename(event_image.filename)
+
+            # Create the full file path for saving the image
             file_path = os.path.join(upload_folder, file_name)
 
+            # Save the uploaded image to the server
             with open(file_path, "wb") as buffer:
```

Ruff (0.8.2)

Async functions should not open files with blocking methods like `open`

(ASYNC230)